api-samples/declarativeNetRequest/no-cookies-manager/manager.js

```diff
@@ -0,0 +1,190 @@
+const ruleItemTemplate = document.getElementById('ruleItemTemplate');
+const rulesList = document.getElementById('rulesList');
+const addRuleButton = document.getElementById('addRuleButton');
+
+async function getNextRuleId() {
+  const rules = await chrome.declarativeNetRequest.getDynamicRules();
+  return Math.max(...rules.map((rule) => rule.id)) + 1;
+}
+
+async function refresh() {
+  const rules = await chrome.declarativeNetRequest.getDynamicRules();
+  renderRules(rules);
+}
+
+refresh();
+
+addRuleButton.addEventListener('click', () => {
+  appendRuleItem(rulesList, {
+    id: 'NEW',
+    conditionType: 'urlFilter',
+    conditionValue: '',
+    caseSensitive: true
+  });
+});
+
+function appendRuleItem(
+  container,
+  { id, conditionType, conditionValue, caseSensitive }
+) {
+  // Create a new rule item from the template
+  let ruleItem = ruleItemTemplate.content.cloneNode(true);
+  container.appendChild(ruleItem);
+  ruleItem = container.lastElementChild;
```

Comment:
Is this needed?

Reply:
I will modify it into the following code:
```js
const ruleItem = ruleItemTemplate.content.cloneNode(true).children[0];
container.appendChild(ruleItem);
```

Because I need to manipulate DOM elements, not the document fragment itself.

Reply:
If that works that seems better 👍